contracts/strategies/PlatypusVoterProxy.sol

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/IPlatypusVoter.sol";
+import "../interfaces/IMasterPlatypus.sol";
+import "../interfaces/IPlatypusPool.sol";
+import "../lib/SafeERC20.sol";
+
+library SafeProxy {
+    function safeExecute(
+        IPlatypusVoter platypusVoter,
+        address target,
+        uint256 value,
+        bytes memory data
+    ) internal returns (bytes memory) {
+        (bool success, bytes memory returnValue) = platypusVoter.execute(target, value, data);
+        if (!success) assert(false);
+        return returnValue;
+    }
+}
+
+contract PlatypusVoterProxy {
+    using SafeMath for uint256;
+    using SafeProxy for IPlatypusVoter;
+    using SafeERC20 for IERC20;
+
+    struct FeeSettings {
+        uint256 stakerFeeBips;
+        uint256 boosterFeeBips;
+        address stakerFeeReceiver;
+        address boosterFeeReceiver;
+    }
+
+    uint256 internal constant BIPS_DIVISOR = 10000;
+
+    uint256 public boosterFee;
+    uint256 public stakerFee;
+    address public stakerFeeReceiver;
+    address public boosterFeeReceiver;
+    address public constant PTP = address(0x22d4002028f537599bE9f666d1c4Fa138522f9c8);
+    IPlatypusVoter public immutable platypusVoter;
+    address public immutable devAddr;
+
+    modifier onlyDev() {
+        require(msg.sender == devAddr, "PlatypusVoterProxy::onlyDev");
+        _;
+    }
+
+    modifier onlyStrategy(address _asset) {
+        require(strategies[_asset] == msg.sender, "PlatypusVoterProxy:onlyStrategy");
+        _;
+    }
+
+    // asset => strategies
+    mapping(address => address) public strategies;
```

Comment:
In theory, there could be a second strategy with the same asset. This mapping means we need to be careful if we redeploy any strategies with the same token. Does not need to be fixed now, but we should document